### ISSUE TITLE
Simple Backpropagation 

### ISSUE THREAD

**Issue by kauecodify** [kauecodify/NeuralLearning.py#2]
```python
import numpy as np

# Função de ativação sigmoid
def sigmoid(x):
    return 1 / (1 + np.exp(-x))

# Derivada da função sigmoid
def sigmoid_derivative(x):
    return x * (1 - x)

# Dados de treinamento
X = np.array([[0, 0], [0, 1], [1, 0], [1, 1]]) # Entradas
y = np.array([[0], [1], [1], [0]]) # Saídas desejadas

# Inicialização dos pesos da rede neural
input_neurons = 2
hidden_neurons = 4
output_neurons = 1

# Pesos da camada de entrada para a camada oculta
weights_input_hidden = np.random.uniform(size=(input_neurons, hidden_neurons))

# Pesos da camada oculta para a camada de saída
weights_hidden_output = np.random.uniform(size=(hidden_neurons, output_neurons))

# Taxa de aprendizado
learning_rate = 0.1

# Treinamento da rede neural
epochs = 10000
for epoch in range(epochs):
    # Forward propagation
    hidden_layer_input = np.dot(X, weights_input_hidden)
    hidden_layer_output = sigmoid(hidden_layer_input)

    output_layer_input = np.dot(hidden_layer_output, weights_hidden_output)
    output_layer_output = sigmoid(output_layer_input)

    # Cálculo do erro
    error = y - output_layer_output

    # Backpropagation
    d_output = error * sigmoid_derivative(output_layer_output)
    error_hidden_layer = d_output.dot(weights_hidden_output.T)
    d_hidden_layer = error_hidden_layer * sigmoid_derivative(hidden_layer_output)

    # Atualização dos pesos
    weights_hidden_output += hidden_layer_output.T.dot(d_output) * learning_rate
    weights_input_hidden += X.T.dot(d_hidden_layer) * learning_rate

# Resultados finais
print("Saída da rede após o treinamento:")
print(output_layer_output)
```

Rede neural com uma camada oculta usando a função de ativação sigmoid. O algoritmo treina a rede neural, operação XOR simples.
